### PR TITLE
feat(hive): add hive dialect

### LINKED ISSUE #8522: Feature:  support hive Dialect
**Summary**

Description for this feature.

### PR DESCRIPTION
**Pull request by sandflee**
I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary

Summary about this PR

Fixes #8522

### CODE REVIEW

File: src/query/settings/src/lib.rs [hunk 1]
```diff
@@ -594,12 +594,10 @@ impl Settings {
         let key = "sql_dialect";
         self.check_and_get_setting_value(key)
             .and_then(|v| v.user_setting.value.as_string())
-            .map(|v| {
-                if v == "MySQL" {
-                    Dialect::MySQL
-                } else {
-                    Dialect::PostgreSQL
-                }
+            .map(|v| match &*v {
```
Comment (Xuanwo) [inline src/query/settings/src/lib.rs:597]: How about converting to lowercase and comparing with `mysql`, `hive` instead?

**Comment by sandflee** [comment]
@merdifly update

**Comment by sandflee** [comment]
@mergify update

File: src/query/settings/src/lib.rs (hunk 1 above)
Comment (sandflee) [reply src/query/settings/src/lib.rs]: done

**Review by Xuanwo — APPROVED** [review]

**Comment by sandflee** [comment]
@mergify update

**Comment by BohuTANG** [comment]
@mergify update

**Review by BohuTANG — APPROVED** [review]

**Comment by Xuanwo** [comment]
@mergify enqueue

**Pull request merged**